uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/request-ctrl_test.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-env amd, jasmine */
```

Comment:
I'd recommend leaving the test in `request-ctrl_test.js` for two reasons:

1. The code that is being tested is on `request-ctrl.js`. It's easier to find the test code for that controller in a file with a similar name, even if it's a parent controller.
2. The code is shared between all child controllers. If for some reason `leave-request-ctrl_test.js` is to be removed, the test for this code will still work ensuring that `sick-request-ctrl.js` and `toil-request-ctrl.js` will work as expected.

Regardless, the better approach is to move all this code into a component so we don't clutter `request-ctrl.js` with extra business logic.